src/Objects/Schema/Generation/SchemaGenerator.php

```diff
@@ -123,7 +123,9 @@ private function schemaFromTypes(Type ...$types): Schema
 
                 return $this->applyAttributes($schema, $attributes);
             default:
-                throw new \InvalidArgumentException('$type is not a valid avro type');
+                $schema = Schema::named($type->getTypeName());
```

Comment:
Do you also have the feeling that this method is doing multiple things - it processes the collection of types and a single type. I believe it makes sense to split these into `schemaFromTypeCollection` and `schemaFromSingleType`.

Additionally I am generally not a huge fan of large `switch`/`case` blocks. One possilbe other way that I have used is using a map of closures instead:

```php
<?php

$typeProcessorMap = [
    TypeName::ENUM => function (SchemaAttributes $attributes) {
        return $this->applyAttributes(Schema::enum(), $attributes);
    },
   // ... more types
];
```

This map can be defined in the constructor of the `SchemaGenerator` class.

Reply:
Idd. I've refactored it and splitted in two methods. Also, I've extracted the mapping logic into a separate class and also followed your suggestion of creating an associative array with closures.